SAT/Z3Interfacing.cpp

```diff
@@ -852,10 +852,15 @@ struct ToZ3Expr
       symb = env.signature->getPredicate(trm->functor());
       range_sort = AtomicSort::boolSort();
       // check for equality
-      if( trm->functor()==0 || symb->equalityProxy()){
+      if( trm->functor()==0 || (symb->equalityProxy())){
         ASS(trm->numTermArguments()==2);
         // both equality and equality proxy translated as z3 equality
-        return args[0] == args[1];
+        if (symb->wasFlipped()) {
```

Comment:
I support this feature and I don't see a better way of doing it. But I will add this to the long list of ways to shoot oneself in the foot, Vampirically.

Reply:
Yes, this can go unnoticed for ages... Unless you sample strategies randomly (sas=z3,ep!=off,rp=on is sufficient) and then prove something really hard! :)